replica/table.cc

```diff
@@ -1904,7 +1904,10 @@ table::make_partition_presence_checker(lw_shared_ptr<sstables::sstable_set> ssta
 snapshot_source
 table::sstables_as_snapshot_source() {
     return snapshot_source([this] () {
-        auto sst_set = _sstables;
+        // Copy the _sstables set to get a snapshot of it
+        // now that table_sstable_set no longer keeps a snapshot
+        // of all compaction_groups sstable_sets
+        auto sst_set = make_lw_shared<sstables::sstable_set>(*_sstables);
```

Comment:
Maybe instead of carefully adjusting select callers, the sstable_set should have a snapshot method. Most implementations will just return a shared_ptr of themselves, while others return a snapshot like the above.

Reply:
We already implement sstable_set copy constructor using a virtual `clone()` method.
v3 of this series has an efficient implementation of it.
We can have a `snapshot` method that will return either `shared_from_this()` or a deep-copy using clone, but I don't think there is a general use case for this yet.  We can add it in the future if needed.